### ISSUE TITLE
FocusZone doesn't restore focus when moving from context menu and item is deleted

### ISSUE THREAD

**Issue by molinger** [microsoft/fluentui#18204]
### Environment Information

- **Package version(s)**: @fluentui/react - 8.14.8
- **Browser and OS versions**: Edge and Windows 10

### Describe the issue:

When the focused element is removed from a list, the focus is moved to the next element. However, if the element is removed while a context menu or dialog box is opened, then focus is not restored to the next element.

#### Repro using context menu:
https://codepen.io/molinger/pen/xxqbyYr

1. Use keyboarding to place focus on a list element.
2. Hit Shift+F10 to open the context menu.
3. Use the Delete context menu item to delete the list item.

#### Repro using dialog:
https://codepen.io/molinger/pen/poeymZZ

1. Use keyboarding to place focus on a list element.
2. Hit Enter to open dialog for deleting element.
3. Hit OK with keyboarding to delete the element and dismiss the dialog.

#### Actual behavior:

Focus is placed on the root of the list.

#### Expected behavior:

Focus should be placed on the item following the one that was deleted since it had focus before the context menu/dialog was opened.

In the context menu case, you can use Enter to delete an element without opening the context menu to show that focus is correctly moved when you aren't using a context menu or dialog.

### Documentation describing expected behavior

https://www.w3.org/WAI/WCAG21/Understanding/focus-order

**Comment by gouttierre** [comment]
@molinger - thank you for filing this issue along with the details into the problem.

@miroslavstastny - Would you be able to confirm if this is a regression, or if this behavior is an issue?
cc @smhigley
